WalletWasabi/Services/WalletService.cs

```diff
@@ -1618,37 +1652,19 @@ private async Task NodeTimeoutsAsync(bool increaseDecrease)
 			Logger.LogInfo<WalletService>($"Current timeout value used on block download is: {timeout} seconds.");
 		}
 
-		#region IDisposable Support
-
-		private volatile bool _disposedValue = false; // To detect redundant calls
-		private Node _localBitcoinCoreNode = null;
-
-		protected virtual void Dispose(bool disposing)
+		public async Task StopAsync()
 		{
-			if (!_disposedValue)
+			while (Interlocked.Read(ref SendCount) != 0) // Make sure to wait for send to finish.
```

Comment:
I think a `ManualResetEvent` could be better than waiting actively in a while loop. Anyway, this will work well of course.